### ISSUE TITLE
Converting tensorflow model to  openvino model error when enable disable_nhwc_to_nchw

### ISSUE THREAD

**Issue by tarekmuallim** [openvinotoolkit/openvino#12419]
I am trying to do a conversion from tensorflow to openvino, but as I enable disable_nhwc_to_nchw I get an error.

If run:
`mo --saved_model_dir /home/tem/scale_model/tensorflow_model --output_dir /home/tem/scale_model/openvino_model --input_shape [1,224,224,3] --data_type FP16 --disable_nhwc_to_nchw --disable_fusing `

The output:
```
home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
[ WARNING ]  Use of deprecated cli option --disable_nhwc_to_nchw detected. Option use in the following releases will be fatal. 
[ WARNING ]  Use of deprecated cli option --disable_fusing detected. Option use in the following releases will be fatal. Please use --finegrain_fusing cli option instead
Model Optimizer arguments:
Common parameters:
	- Path to the Input Model: 	None
	- Path for generated IR: 	/home/tem/scale_model/openvino_model
	- IR output name: 	saved_model
	- Log level: 	ERROR
	- Batch: 	Not specified, inherited from the model
	- Input layers: 	Not specified, inherited from the model
	- Output layers: 	Not specified, inherited from the model
	- Input shapes: 	[1,224,224,3]
	- Source layout: 	Not specified
	- Target layout: 	Not specified
	- Layout: 	Not specified
	- Mean values: 	Not specified
	- Scale values: 	Not specified
	- Scale factor: 	Not specified
	- Precision of IR: 	FP16
	- Enable fusing: 	False
	- User transformations: 	Not specified
	- Reverse input channels: 	False
	- Enable IR generation for fixed input shape: 	False
	- Use the transformations config file: 	None
Advanced parameters:
	- Force the usage of legacy Frontend of Model Optimizer for model conversion into IR: 	False
	- Force the usage of new Frontend of Model Optimizer for model conversion into IR: 	False
TensorFlow specific parameters:
	- Input model in text protobuf format: 	False
	- Path to model dump for TensorBoard: 	None
	- List of shared libraries with TensorFlow custom layers implementation: 	None
	- Update the configuration file with input/output node names: 	None
	- Use configuration file used to generate the model with Object Detection API: 	None
	- Use the config file: 	None
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
OpenVINO runtime found in: 	/home/tem/intel/openvino_2022/python/python3.8/openvino
OpenVINO runtime version: 	2022.1.0-7019-cdb9bec7210-releases/2022/1
Model Optimizer version: 	2022.1.0-7019-cdb9bec7210-releases/2022/1
[ ERROR ]  Exception occurred during running replacer "fusing" (<class 'openvino.tools.mo.middle.fusings.Fusing'>): Input shapes mismatch for node StatefulPartitionedCall/sequential/MobileNetV2Base/bn_Conv1/FusedBatchNormV3/mean: [masked_array(data=[  1, 112, 112,  32],
             mask=False,
       fill_value=-1000000007), array([32,  1,  1])]
```
If run:
`mo --saved_model_dir /home/tem/scale_model/tensorflow_model --output_dir /home/tem/scale_model/openvino_model --input_shape [1,224,224,3] --data_type FP16 --disable_fusing`

The output:
```
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
[ WARNING ]  Use of deprecated cli option --disable_fusing detected. Option use in the following releases will be fatal. Please use --finegrain_fusing cli option instead
Model Optimizer arguments:
Common parameters:
	- Path to the Input Model: 	None
	- Path for generated IR: 	/home/tem/scale_model/openvino_model
	- IR output name: 	saved_model
	- Log level: 	ERROR
	- Batch: 	Not specified, inherited from the model
	- Input layers: 	Not specified, inherited from the model
	- Output layers: 	Not specified, inherited from the model
	- Input shapes: 	[1,224,224,3]
	- Source layout: 	Not specified
	- Target layout: 	Not specified
	- Layout: 	Not specified
	- Mean values: 	Not specified
	- Scale values: 	Not specified
	- Scale factor: 	Not specified
	- Precision of IR: 	FP16
	- Enable fusing: 	False
	- User transformations: 	Not specified
	- Reverse input channels: 	False
	- Enable IR generation for fixed input shape: 	False
	- Use the transformations config file: 	None
Advanced parameters:
	- Force the usage of legacy Frontend of Model Optimizer for model conversion into IR: 	False
	- Force the usage of new Frontend of Model Optimizer for model conversion into IR: 	False
TensorFlow specific parameters:
	- Input model in text protobuf format: 	False
	- Path to model dump for TensorBoard: 	None
	- List of shared libraries with TensorFlow custom layers implementation: 	None
	- Update the configuration file with input/output node names: 	None
	- Use configuration file used to generate the model with Object Detection API: 	None
	- Use the config file: 	None
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
OpenVINO runtime found in: 	/home/tem/intel/openvino_2022/python/python3.8/openvino
OpenVINO runtime version: 	2022.1.0-7019-cdb9bec7210-releases/2022/1
Model Optimizer version: 	2022.1.0-7019-cdb9bec7210-releases/2022/1
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/tem/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
[ SUCCESS ] Generated IR version 11 model.
[ SUCCESS ] XML file: /home/tem/scale_model/openvino_model/saved_model.xml
[ SUCCESS ] BIN file: /home/tem/scale_model/openvino_model/saved_model.bin
[ SUCCESS ] Total execution time: 21.06 seconds. 
[ SUCCESS ] Memory consumed: 2483 MB. 
It's been a while, check for a new version of Intel(R) Distribution of OpenVINO(TM) toolkit here https://software.intel.com/content/www/us/en/develop/tools/openvino-toolkit/download.html?cid=other&source=prod&campid=ww_2022_bu_IOTG_OpenVINO-2022-1&content=upg_all&medium=organic or on the GitHub*
[ INFO ] The model was converted to IR v11, the latest model format that corresponds to the source DL framework input/output format. While IR v11 is backwards compatible with OpenVINO Inference Engine API v1.0, please use API v2.0 (as of 2022.1) to take advantage of the latest improvements in IR v11.
Find more information about API v2.0 and IR v11 at https://docs.openvino.ai
```

**Comment by zulkifli-halim** [comment]
Hi @tarekmuallim.

If your TF model is in NCHW layout you need to specify `--disable_nhwc_to_nchw` command line parameter when you convert the model. The `--disable_nhwc_to_nchw` parameter will disable the default translation from NHWC to NCHW. 

From the error message during optimization, when you specifying the `--disable_nhwc_to_nchw` this will cause a mismatch in the input shape.

For your information, starting from OpenVINO 2022.1, automatic detection of `--disable_nhwc_to_nchw`  was implemented. Therefore, this parameter is no longer needed. 

You can refer to release notes for [more info](https://www.intel.com/content/www/us/en/developer/articles/release-notes/openvino-relnotes.html#inpage-nav-3-undefined).

**Comment by tarekmuallim** [comment]
Hi @zulkifli-halim

Thank you for the detailed information.

Like that, the converted model will be NCHW all that time.
What if I want the converted model to be NHWC instead. Is there a way to do that?

**Comment by zulkifli-halim** [comment]
@tarekmuallim. You can refer to [When to specify Layout](https://docs.openvino.ai/latest/openvino_docs_MO_DG_Additional_Optimization_Use_Cases.html#when-to-specify-layout) and [How to Change Layout of a Model Inputs and Outputs](https://docs.openvino.ai/latest/openvino_docs_MO_DG_Additional_Optimization_Use_Cases.html#how-to-change-layout-of-a-model-inputs-and-outputs) documentation for more detail.

**Comment by tarekmuallim** [comment]
Hi @zulkifli-halim 

Thank you very much for the information.
I trying some thing like:

`mo --input_model /home/tem/Fruits-Vegetable-Inference-OAK/frozen_model.pb --output_dir /home/tem/Fruits-Vegetable-Inference-OAK/openvino_model --input_shape [1,224,224,3] --data_type FP16 --disable_fusing --layout MobileNetV2Base_input[n,h,w,c],dense_2/Softmax[n,c]`

but I am getting an error:

`[ ERROR ]  Input/Output with name dense_2/Softmax wasn't found! `

I am pretty sure of the output name.

**Comment by zulkifli-halim** [comment]
Hi @tarekmuallim, can you share your model with me?

**Comment by tarekmuallim** [comment]
Hi @zulkifli-halim 

I solve the problem.
I used:
`mo --input_model /home/tem/Fruits-Vegetable-Inference-OAK/frozen_model.pb --output_dir /home/tem/Fruits-Vegetable-Inference-OAK/openvino_model --input_shape [1,224,224,3] --data_type FP16 --disable_fusing --layout MobileNetV2Base_input[n,h,w,c],dense_2/Softmax:0[n,c]`

I had to add `:0` to the output name.

Thank you for your help.
